src/lingrex/trimming.py

```diff
@@ -1,201 +1,178 @@
+        return [s.soundclass(gap=self.gap) for s in self]
+
+    def trimmed(self, idxs: typing.Iterable[int]) -> 'Sites':
+        for idx in sorted(set(idxs), reverse=True):
+            del self[idx]
+        return self
+
+    def to_alignment(self) -> typing.List[typing.List[str]]:
+        return [[s[i] for s in self] for i in range(len(self[0]))]
+
+    def trimmed_by_gap(self,
+                       threshold: float = 0.5,
+                       skeletons: typing.Iterable[str] = ("CV", "VC"),
+                       exclude="_+") -> 'Sites':
+        """
+        Trim alignment sites by gaps.
+
+        :parameter threshold: Threshold for gap ratio by which sites should be trimmed.
+        :param skeletons: Iterable of syllable-skeletons at least one of which should be preserved \
+        for further processing.
+        :param exclude: Sequence of strings that should be excluded from further processing,
+            e.g. morpheme boundaries. Defaults to '_+'.
+        """
+        skeleton = list(enumerate(self.soundclasses))
+
+        # exclude markers
+        idxs = [i for i, c in skeleton if c in exclude]
+        # order by gap weights
+        for idx, score in sorted(enumerate(self.gap_ratios), key=lambda x: x[1], reverse=True):
+            if score >= threshold:
+                current_skeleton = [c for i, c in skeleton if i not in idxs + [idx]]
+                if any([subsequence_of(s, current_skeleton) for s in skeletons]):
+                    # Trimming this site leaves a "big enough" remainder.
+                    idxs.append(idx)
+                else:
+                    break
+            else:
+                break
+        return self.trimmed(idxs)
+
+    def trimmed_by_core(self,
+                        threshold: float = 0.5,
+                        skeletons: typing.Iterable[str] = ("CV", "VC"),
+                        exclude="_+") -> 'Sites':
+        """
+        Trim alignment sites by gaps, preserving a core of sites.
+
+        :parameter threshold: Threshold by which sites with gaps should be trimmed.
+        :param skeletons: Tuple of syllable-skeletons that should be preserved
+            for further processing. Defaults to '("CV", "VC")'.
+        :parameter gap: String that codes gaps in alignment sites. Defaults to '-'.
+        """
+        skeleton = list(enumerate(self.soundclasses))
+        cons = [self.gap if ratio >= threshold else "S" for ratio in self.gap_ratios]
+        takewhile_gap = functools.partial(itertools.takewhile, lambda c: c[1] == self.gap)
+        left = [i for i, _ in takewhile_gap(enumerate(cons))]
+        right = [len(cons) - 1 - i for i, _ in takewhile_gap(enumerate(reversed(cons)))]
+
+        idxs = [i for i, c in skeleton if c in exclude]
+
+        for idx in right + left:
+            current_skeleton = [c for i, c in skeleton if i not in idxs + [idx]]
             if any([subsequence_of(s, current_skeleton) for s in skeletons]):
```

Comment:
A note on `any`: To make `any` efficient, it should be used with a generator as argument:
```python
>>> any([1/x for x in [3,2,0]])
Traceback (most recent call last):
ZeroDivisionError: division by zero
>>> any(1/x for x in [3,2,0])
True
```
In the second case, `any` returns as soon as the first true value is found.